### PR TITLE
Rework rendering - render planes separately

### PR DESCRIPTION
**Pull request by WojciechBarczynski**
Should remove raise conditions in rendering and fix sampling transparent textures

### CODE REVIEW

File: compositor_render/src/transformations/layout/shader.rs [hunk 1]
```diff
@@ -145,9 +145,12 @@ impl LayoutShader {
             render_pass.set_bind_group(1, params, &[]);
             render_pass.set_bind_group(2, &self.sampler.bind_group, &[]);
 
-            self.surfaces.draw(&mut render_pass, layout_count);
+            wgpu_ctx
+                .planes
+                .get_plane(layout_id as i32)
+                .unwrap()
```
[the remark this answers is not in the document]
Comment (wkozyra95) [reply compositor_render/src/transformations/layout/shader.rs:150]: In the old implementation we were limited by a shader source, but with new approach you don't have that problem, so if we can relatively easily support infinite amount of layouts then we should. (not necessarily in this PR)

**Review by noituri — APPROVED** [review]
LGTM

**Comment by wkozyra95** [comment]
new-api was merged to master, rebase this PR

File: compositor_render/src/wgpu/shader/pipeline.rs [hunk 2]
```diff
@@ -154,8 +153,19 @@ impl Pipeline {
...
+                .draw(&mut render_pass);
+        };
+
+        if common_parameters.texture_count == 0 {
+            render_plane(-1, true);
+        } else {
+            for input_id in 0..common_parameters.texture_count {
+                let is_first = input_id == 0;
+                render_plane(input_id as i32, is_first);
+            }
```
Comment (wkozyra95) [inline compositor_render/src/wgpu/shader/pipeline.rs:168]:
```suggestion
            render_plane(0, true);
            for input_id in 1..common_parameters.texture_count {
                render_plane(input_id as i32, false);
            }
```

**Review by wkozyra95 — APPROVED** [review]
Confirm no aritfacts after this change.

20231211_10h51m21s_grim

**Pull request merged**